### DISCUSSION TITLE
generic responses?

### DISCUSSION THREAD

**Discussion by amirongit** [Neoteroi/BlackSheep#420] [Ideas]
so you could specify the structure of your http response in the signature of request handler like `Response[MyEntity | YourErrors]`.

two benefits:

the document generator could find the description and generate documentation from the signature of the request handler

the developer would have the ability to customize and document the structure of errors and tweak the response object with custom status codes and other properties without needing to document it manually

**Comment by amirongit** [comment]
I don't like this idea anymore

**Reply by RobertoPrevato** [reply]
@amirongit I considered that, and I like the idea. I tried to implement in the past and for some reason that I don´t recall now, I abandoned that idea. However, I would like to try supporting more details using `Annotated`.
